checkov/secrets/scan_git_history.py

```diff
@@ -1,42 +1,39 @@
 from __future__ import annotations
 
-import copy
 import logging
 import os
-from collections import defaultdict
-from typing import TYPE_CHECKING, Dict, List, Tuple, Optional
+
+from typing import TYPE_CHECKING, Dict, Optional
 from checkov.common.util import stopit
 from detect_secrets.core import scan
-from typing_extensions import TypedDict
 
+from checkov.secrets.git_history_store import GitHistorySecretStore
 from checkov.secrets.consts import GIT_HISTORY_NOT_BEEN_REMOVED
 
 if TYPE_CHECKING:
     from detect_secrets import SecretsCollection
-    from detect_secrets.core.potential_secret import PotentialSecret
 
 os.environ["GIT_PYTHON_REFRESH"] = "quiet"
 try:
     import git
-
     git_import_error = None
 except ImportError as e:
     git_import_error = e
 
-
-class EnrichedPotentialSecret(TypedDict):
-    added_commit_hash: str
-    removed_commit_hash: str
-    potential_secret: PotentialSecret
-    code_line: Optional[str]
+GIT_HISTORY_SEC_STORE = GitHistorySecretStore()
 
 
 class GitHistoryScanner:
-    def __init__(self, root_folder: str, secrets: SecretsCollection, timeout: int = 43200):
+    def __init__(self, root_folder: str, secrets: SecretsCollection,
+                 secret_store: GitHistorySecretStore = GIT_HISTORY_SEC_STORE, timeout: int = 43200):
         self.root_folder = root_folder
         self.secrets = secrets
         self.timeout = timeout
-        self.secret_store = GitHistorySecretStore()
+        # in case we start from mid-history (git) we want to continue from where we've been
+        if secret_store:
+            self.secret_store = secret_store
+        else:
+            self.secret_store = GitHistorySecretStore()
```

Comment:
can be simplified to
```suggestion
        self.secret_store = secret_store or GitHistorySecretStore()
```

Reply:
cool